app/components/champ-selection.tsx

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useState } from "react";
+import ChampionsClient from "../_clients/champions-client/champions-client";
+import ChampCard from "./champ-card";
+import {
+  Box,
+  Button,
+  FormControl,
+  FormLabel,
+  HStack,
+  Input,
+} from "@chakra-ui/react";
+
+function ChampSelection({
+  onChampSelection,
+}: {
+  onChampSelection: (championData: ChampionData) => void;
+}) {
+  const [champions, setChampions] = useState<ChampionData[]>(
+    ChampionsClient.getChampionsArray(),
+  );
+
+  const [champSearchValue, setChampionSearchValue] = useState<string>("");
```

Comment:
No need to define type if you pass a type primitive as the initial state. 
```suggestion
  const [champSearchValue, setChampionSearchValue] = useState("");
```